From 76360812ca4d71b0586b92009c90ba741ffccf3f Mon Sep 17 00:00:00 2001
Message-Id: <76360812ca4d71b0586b92009c90ba741ffccf3f.1374754301.git.minovotn@redhat.com>
In-Reply-To: <5d75a8513d08b33975bdf5971871c0c977167cd1.1374754301.git.minovotn@redhat.com>
References: <5d75a8513d08b33975bdf5971871c0c977167cd1.1374754301.git.minovotn@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Mon, 24 Jun 2013 07:05:25 +0200
Subject: [PATCH 14/65] qemu-sockets: unix_listen and unix_connect are
 portable

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1372057576-26450-15-git-send-email-kraxel@redhat.com>
Patchwork-id: 52138
O-Subject: [RHEL-6.5 qemu-kvm PATCH v2 14/65] qemu-sockets: unix_listen and unix_connect are portable
Bugzilla: 676568
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

They are just wrappers and do not need a Win32-specific version.

Reviewed-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 0c814709471ac8a9617406e1e2cbd7015e97bfe9)

Conflicts:

	qemu-sockets.c
---
 qemu-sockets.c |   45 +++++++++++++++++----------------------------
 1 file changed, 17 insertions(+), 28 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 qemu-sockets.c | 45 +++++++++++++++++----------------------------
 1 file changed, 17 insertions(+), 28 deletions(-)

diff --git a/qemu-sockets.c b/qemu-sockets.c
index 341f175..ff80e6a 100644
--- a/qemu-sockets.c
+++ b/qemu-sockets.c
@@ -732,6 +732,23 @@ int unix_connect_opts(QemuOpts *opts, Error **errp)
     return sock;
 }
 
+#else
+
+int unix_listen_opts(QemuOpts *opts, Error **errp)
+{
+    fprintf(stderr, "unix sockets are not available on windows\n");
+    errno = ENOTSUP;
+    return -1;
+}
+
+int unix_connect_opts(QemuOpts *opts, Error **errp)
+{
+    fprintf(stderr, "unix sockets are not available on windows\n");
+    errno = ENOTSUP;
+    return -1;
+}
+#endif
+
 /* compatibility wrapper */
 int unix_listen(const char *str, char *ostr, int olen, Error **errp)
 {
@@ -773,31 +790,3 @@ int unix_connect(const char *path, Error **errp)
     qemu_opts_del(opts);
     return sock;
 }
-
-#else
-
-int unix_listen_opts(QemuOpts *opts, Error **errp)
-{
-    fprintf(stderr, "unix sockets are not available on windows\n");
-    return -1;
-}
-
-int unix_connect_opts(QemuOpts *opts, Error **errp)
-{
-    fprintf(stderr, "unix sockets are not available on windows\n");
-    return -1;
-}
-
-int unix_listen(const char *path, char *ostr, int olen, Error **errp)
-{
-    fprintf(stderr, "unix sockets are not available on windows\n");
-    return -1;
-}
-
-int unix_connect(const char *path, Error **errp)
-{
-    fprintf(stderr, "unix sockets are not available on windows\n");
-    return -1;
-}
-
-#endif
-- 
1.7.11.7

